src/fu-install-task.c

```diff
@@ -217,6 +232,20 @@ fu_install_task_check_requirements (FuInstallTask *self,
 		return FALSE;
 	}
 
+	/* check the version formats match */
+	fmt = fu_install_task_guess_version_format (self, version_release);
+	if (fmt != fu_device_get_version_format (self->device) &&
+	    (flags & FWUPD_INSTALL_FLAG_FORCE) == 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "Firmware version formats were different, "
+			     "device was '%s' and release is '%s'",
+			     fu_common_version_format_to_string (fu_device_get_version_format (self->device)),
+			     fu_common_version_format_to_string (fmt));
```

Comment:
I tried this on the problematic device related to activation.

```
Firmware version formats were different, device was '(null)' and release is 'plain'
```

Which doesn't seem right - in fu_device_set_version the format definitely gets guessed into plain.

Reply:
It looks like to me the incorporate method never incorporates version format, so probably that's the cause of this.

Reply:
I pushed a fix for this.